cryptography/src/secp256r1/scheme.rs

```diff
@@ -503,14 +505,14 @@ mod tests {
         let encoded = original.encode();
         assert_eq!(encoded.len(), SIGNATURE_LENGTH);
 
-        let decoded = Signature::decode(encoded).unwrap();
+        let decoded = Signature::decode::<Bytes>(encoded).unwrap();
```

Comment:
I'm surprised there is no way to infer this from the provided argument?

Reply:
I wonder if we could just do `fn write(&self, buf: &mut impl BufMut)`?

Reply:
Don't quite understand your comment. In the current changes, we have ```fn write<B: BufMut>(&self, buf: &mut B)```, which I think is the same as what you suggested ```fn write(&self, buf: &mut impl BufMut)``` just a different syntax?